### DISCUSSION TITLE
Would love to be able to automatically hide diffs that only show import changes

### DISCUSSION THREAD

**Discussion by shawn-rice-m43** [community/community#16102] [Pull Requests]
My organization is in the process of deduplicating shared code, which results in large PRs with hundreds of changed files, most of which are just retrieving the changed module from its new home in the project.

We would *love* to be able to opt-in to having all file changes in a PR that only have line changes that begin with the `import` keyword to be automatically collapsed, perhaps with an indicator that it's an import-only change.

I'm curious to see if there are others in the GitHub community that feels the same way.

**Comment by chuck0523** [comment]
I'm in a similar situation.
Sometime, PR contains dozens of file changes, especially refactoring one. And I don't want reviewers to consume too much time on it.

It's highly appreciated it we can have this feature! 
Could be named like "Mark as important change", maybe?

**Reply by shawn-rice-m43** [reply]
That's a great suggestion!

**Comment by davidhatanian-stripe** [comment]
Same here! Imports are automatically enforced by our build system and for some changes files with only import changes represent 50% of the PR, which I'd rather not see.

I'd very much appreciate "partial hiding" where import changes are hidden on all files, even files that have non-import changes. But even a blanket "hide files when the only changes are imports" would be a huge help.

One alternative would be to make the hiding logic dynamic, eg through a Github action, so that GH users can write their own language and context-specific hiding logic.

**Comment by JongSikLim** [comment]
Same experience! My team can't share a good review if there's a big rename diff in the PR.
It would be good to handle the visibility of rename diff using toggle.

If it is such a challenging suggestion, could you improve the performance of the diff page? This point needs more than just this case.
